tests/verify.test.js

```diff
@@ -0,0 +1,89 @@
+import chai from 'chai';
+import chaihttp from 'chai-http';
+import createToken from '../src/helpers/tokens/generate.token';
+import env from '../src/configs/environments';
+import server from '../src/index';
+import status from '../src/helpers/constants/status.codes';
+
+
+chai.use(chaihttp);
+chai.should();
+
+
+describe('GET /verifyEmail', () => {
+  it('Should return a success message if the link is valid', (done) => {
+    const user = {
+      username: 'BurindiAlain',
+      email: 'alain@gmail.com',
+      password: 'password23423',
+      confirmPassword: 'password23423'
+    };
+
+    const { username, email } = user;
+
+    const verifytoken = createToken({ username, email }, env.secret);
+    chai
+      .request(server)
+      .get(`/api/users/verifyEmail/${verifytoken}`)
+      .end(async (err, res) => {
+        await res.should.have.status(status.OK);
+        await res.body.message.should.be.eq('Your email is successfully verified');
+        done();
+      });
+  });
+
+  it('Should return a message if a user doesn\'t exist', (done) => {
+    const user = {
+      username: 'BurindiAlain',
+      email: 'alain@gmail.com',
+      password: 'password23423',
+      confirmPassword: 'password23423'
+    };
+    chai
+      .request(server)
+      .post('/api/users/signup')
+      .send(user)
+      .end();
+
+    const { username } = user;
+
+    const verifytoken = createToken({ username, email: 'dummy@email.com' }, env.secret);
+
+    chai
+      .request(server)
+      .get(`/api/users/verifyEmail/${verifytoken}`)
+      .end((err, res) => {
+        res.should.have.status(status.NOT_FOUND);
+        res.body.message.should.be.eq('User doesn\'t exist.');
+        done();
+      });
+  });
+
+  it('Should respond with a message if the token is invalid', (done) => {
+    const user = {
+      username: 'BurindiAlain',
+      email: 'alain@gmail.com',
+      password: 'password23423',
+      confirmPassword: 'password23423'
+    };
+
+    chai
+      .request(server)
+      .post('/api/users/signup')
+      .send(user)
+      .end();
+
+    const { username } = user;
+
+    const verifytoken = createToken({ username }, 'env.secret');
+
+    chai
+      .request(server)
+      .get(`/api/users/verifyEmail/${verifytoken}`)
+      .end((err, res) => {
+        res.should.have.status(status.BAD_REQUEST);
+        res.body.message.should.be.eq('The link provided is  corrupt, please request a new one or try to click it again');
+        done();
```

Comment:
same on this file use error.sender.js

Reply:
@truestbyheart have you implemented @murediane's feedback above?

Reply:
yes,  I have implemented it @MemunaHaruna